src/file-generator.js

```diff
@@ -0,0 +1,18 @@
+const fs = require("fs");
+
+module.exports = (outDir, output) => {
+    try {
+        try {
+            fs.lstatSync(outDir).isDirectory();
+        } catch(e) {
+            console.warning("Output dir doesnt exists. Creating it...");
+            fs.mkdirSync(outDir, '0777');
+        } finally {
+            fs.writeFileSync(outDir + "budget.html", output);
+            return true;
+        }
+    } catch (e) {
+        console.log(e);
+        return false;
+    }
```

Comment:
Really needs to catch a exception to check if the directory exists? I checked the docs and found a function that returns a `boolean` and looks goods

https://nodejs.org/api/fs.html#fs_fs_existssync_path

We can do something like that?
```javascript
const fs from 'fs';

try {
if (!fs.existsSync(outDir)) {
  console.warning("Output dir doesnt exists. Creating it...");
  fs.mkdirSync(outDir, '0777');
}
fs.writeFileSync(outDir + "budget.html", output);
} catch (e) {
        console.log(e);
        return false;
    }
.
```

Reply:
Nice, i was not sure if that exists and didnt checked the docs. I ll implement this way